lib/ansible/plugins/lookup/sops.py

```diff
@@ -0,0 +1,171 @@
+            description: determine if sops output should be decoded as text or considered binary
+            type: bool
+            required: False
+            default: False
+        print:
+            description: print the decrypted content
+            type: bool
+            required: False
+            default: False
+    notes:
+        - this lookup does not understand 'globing' - use the fileglob lookup instead.
+"""
+
+EXAMPLES = """
+tasks:
+  - name : output secrets to screen (BAD IDEA)
+    debug:
+        msg: "Content: {{lookup('sops', item)}}"
+    with_file:
+    - sops-encrypted-file.enc.yaml
+
+  - name: add ssh private key
+    copy:
+        content: "{{lookup('sops', user + '-id_rsa')}}"
+        dest: /home/{{user}}/.ssh/id_rsa
+        owner: "{{user}}"
+        group: "{{user}}"
+        mode: 0600
+"""
+
+RETURN = """
+    _raw:
+        description: decrypted file content
+"""
+
+# From https://github.com/mozilla/sops/blob/master/cmd/sops/codes/codes.go
+# Should be manually updated
+sops_error_codes = {
+    1: "ErrorGeneric",
+    2: "CouldNotReadInputFile",
+    3: "CouldNotWriteOutputFile",
+    4: "ErrorDumpingTree",
+    5: "ErrorReadingConfig",
+    6: "ErrorInvalidKMSEncryptionContextFormat",
+    7: "ErrorInvalidSetFormat",
+    8: "ErrorConflictingParameters",
+    21: "ErrorEncryptingMac",
+    23: "ErrorEncryptingTree",
+    24: "ErrorDecryptingMac",
+    25: "ErrorDecryptingTree",
+    49: "CannotChangeKeysFromNonExistentFile",
+    51: "MacMismatch",
+    52: "MacNotFound",
+    61: "ConfigFileNotFound",
+    85: "KeyboardInterrupt",
+    91: "InvalidTreePathFormat",
+    100: "NoFileSpecified",
+    128: "CouldNotRetrieveKey",
+    111: "NoEncryptionKeyFound",
+    200: "FileHasNotBeenModified",
+    201: "NoEditorFound",
+    202: "FailedToCompareVersions",
+    203: "FileAlreadyEncrypted"
+}
+
+
+class Sops():
+    @staticmethod
+    def decrypt(encrypted_file):
+        # Run sops directly, python module is deprecated
+        command = ["sops", "--decrypt", encrypted_file]
+        process = Popen(command, stdout=PIPE, stderr=PIPE)
+        (output, err) = process.communicate()
+        exit_code = process.wait()
+        return output, err, exit_code
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+        ret = []
+
+        sops_text_output = True
+        if 'binary' in kwargs:
+            sops_text_output = not kwargs['binary']
+
+        print_decrypted = False
+        if 'print' in kwargs:
+            print_decrypted = kwargs['print']
+
+        for term in terms:
+            display.debug("Sops lookup term: %s" % term)
+            lookupfile = self.find_file_in_search_path(variables, 'files', term)
+            display.vvvv(u"Sops lookup using %s as file" % lookupfile)
+
+            try:
+                if lookupfile:
+                    (output, err, exit_code) = Sops.decrypt(lookupfile)
+
+                    if sops_text_output:
```

Comment:
Lookups should always return "text".  As such, you should make use of `ansible.module_utils._text.to_text` to handle enforce the type.
```
output = to_text(output, errors='surrogate_or_strict')
```

Reply:
thanks for the suggestion! I changed the code to use the proper ansible function.